internal/injector/config/config-go.go

```diff
@@ -0,0 +1,141 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/DataDog/orchestrion/internal/injector/aspect"
+	"golang.org/x/tools/go/packages"
+)
+
+const OrchestrionToolGo = "orchestrion.tool.go"
+
+var ErrInvalidGoPackage = errors.New("no .go files in package")
+
+// loadGoPackage loads configuration from the specified go package.
+func (l *Loader) loadGoPackage(pkg *packages.Package) (*configGo, error) {
+	root := packageRoot(pkg)
+	if root == "" {
+		// This might be explained by a package-level loading error... We only check
+		// here because "all .go files are excluded by build constraints" is one
+		// such error that we typically ignore.
+		if pkg.Errors != nil {
+			var err error
+			for _, e := range pkg.Errors {
+				err = errors.Join(err, e)
+			}
+			return nil, fmt.Errorf("in %q: %w", pkg.ID, err)
+		}
+
+		return nil, fmt.Errorf("%q: %w", pkg.PkgPath, ErrInvalidGoPackage)
+	}
+
+	toolFile := filepath.Join(root, OrchestrionToolGo)
+	imports, err := l.loadGoFile(toolFile)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return nil, err
+	}
+
+	ymlCfg, err := l.loadYMLFile(root, OrchestrionYML)
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return nil, err
+	}
+
+	return &configGo{imports: imports, yaml: ymlCfg, pkgPath: pkg.PkgPath}, nil
+}
+
+// loadGoFile loads configuration from the specified go file. Returns nil if the
+// file had already been loaded previously, or if the configuration is empty.
+func (l *Loader) loadGoFile(filename string) ([]Config, error) {
+	if !l.markLoaded(filename) {
+		return nil, nil
+	}
```

Comment:
I believe this could cause a issue because `loadGoPackage` does not handle the case where `loadGoFile` returns `nil, nil`. I believe putting the `markLoaded` on packages paths instead of filenames would make us able to make it go higher in the call stack and make things easier to manage.

Reply:
If `loadGoFile` returns `nil, nil`, the slice of "imported packages" is `nil` which is the same as an empty slice... So that does not need to be handled specifically.

Doing `markLoaded` on package paths would work if the `orchestrion.yml` files were only able to refer to go packages, but they're able to refer to specific YAML documents with an arbitrary filename, and we still want to load each such YAML document exactly once.